### ISSUE TITLE
Game Shortcut of Non-Steam games added to Steam

### ISSUE THREAD

**Issue by Smoukus** [terrymacdonald/DisplayMagician#275]
Hey, I'm writing here because I haven't found any contact link / info other than the issue tracker.

First of all, thanks a lot for writing this piece of software. It is amazing, and made my workflow 10x better and efficient!

Here comes my problem:
I have bunch of games added to Steam via "add non-steam games" setting in order to get Steam Input functionality. If I don't do that, games don't recognise my controller input.

Is it somehow possible to create DisplayMagician Game Shortcut from those Non-Steam Game Shortcuts?
They look like this for example:
![image](https://github.com/terrymacdonald/DisplayMagician/assets/48409293/a0cf6535-8d65-4634-949c-54d91e2dae49)

![image](https://github.com/terrymacdonald/DisplayMagician/assets/48409293/80e59924-e965-4a86-99c5-66e99592fbd2)

Is something like that possible?

**Comment by terrymacdonald** [comment]
Hi @Smoukus. This is exactly the right way to add feature requests and enhancements.

Thanks so much for the kind words. I get a big kick from people making use of my hard work, so it's lovely to have such nice feedback.

I'd have to do some testing in order to see how Steam handles those 3rd party apps. I've never used that feature before so I don't really have any understanding of how difficult it would be to do. As long as Steam provides a URL way of running the app (which it appears to do from the screenshots you posted) then I should be able to do something with them. We'll have to wait and see.

I don't have as much time to work on this as I have in the past unfortunately due to life, but I'll keep this in the backlog as I think it shouldn't be too difficult to add..

Thanks
Terry

**Comment by Smoukus** [comment]
Hey Terry,

thanks for your reply. I'm glad that there's hope for this functionality after reading your answer :)
In case you need help with testing or providing any additional information regarding this specific topic, just let me know.

I tried to add that Steam created shortcut into Display Magician as an executable, however it hadn't worked. I pressume that it hadn't worked because the Steam's shortcut has an .url extension.

Thanks in advance!

**Comment by Smoukus** [comment]
Just wanted to ask if this feature would be added with the 3.0 release, or could it be added before it?
As I've read in your other issues, 3.0 release is a huge rewrite of the entire codebase, and it's pretty time consuming.
Therefore it might take quite some time before the 3.0 releases alongside this feature.

**Comment by terrymacdonald** [comment]
@Smoukus I've finally had a bit of time to review this. I want you to know it's not a simple addition. There is quite a bit of research and testing that needs to go into producing this. I need to figure out the structure changes that Steam have made in order to add this feature, then understand how to extract the information I need for getting the Game Shortcuts. I then need to write code to be able to do this in a reliable fashion. It all takes a lot of effort.

I have spent an hour or so looking for where the data is written. It appears to be stored in the C:\Program Files (x86)\Steam\userdata\<UserID>\config` folder in the shortcuts.vdf file. It seems like it is set per user, which complicates thing a bit more. It is a VDF file which we already can import thanks to prior work, but the impact of per user settings needs to be investigated more. 

I'll have a think about how much time this will take, and then I'll figure out if I'll be able to do it within the next release  or not. The next release v2.6.0 is due soon, so I may not have time to get this in before that ships. There are a few critical improvements to video handling in v2.6.0 which means I need to get it out sooner rather than later.

I'll drop more notes here as I complete my research so I don't lose them, and hopefully they may help others on the internet wanting to add their own similar feature.

Thanks
Terry

**Comment by Smoukus** [comment]
Hey @terrymacdonald, thanks a bunch for still looking into this and taking your time. If you need help with testing, or help in some other way, just let me know.

**Comment by Smoukus** [comment]
Any progress on this topic? This is such a game changer for people who wish to play non Steam games with a controller.

**Comment by terrymacdonald** [comment]
Hi @Smoukus,


Unfortunately no progress. I've been working on fixing a few critical bugs
in the video library processing, which has resulted in a need to rewrite
the NVIDIA library. This is going to take ages, but it is a critical
component of making DisplayMagician reliable.

I'm not 100% convinced that supporting non Steam games from steam will
impact that many people as they can always create an application executable
shortcut manually if needed. I tend to work on issues that have the widest
impact to my user base first, and those issues without a work around next.

It may be a while longer before I'll be in a position to add this feature.

Thanks
Terry

**Comment by Smoukus** [comment]
Thanks for the reply Terry. I understand what you mean and that's alright. Thank you for your time into developing this app regardless.

I'd like to comment on the following:
> I'm not 100% convinced that supporting non Steam games from steam will
impact that many people as they can always create an application executable
shortcut manually if needed.

That is true, however that way you rely entirely on application's support of the controller. Steam Input doesn't work unless the game has been added to Steam, and launched via Steam or via Steam-created shortcut.

There are bunch of games that have only Xbox Controller native support. If you wish to use Playstation Controller for example in such game, you need to enable Steam Input for it to work.
If you have that game on Steam, it's not a problem (works already via Display Magician perfectly). However if you don't have the game on Steam, the only way for you to get the Controller working is to add the game into Steam and launch it through there.

**Comment by terrymacdonald** [comment]
Hi Smoukus,

That's a good point about the controllers. I'll see what I can do after
I've fixed the NVIDIA issues. No promises though as Steam may resist my
efforts to get this to work! They definitely don't make it easy.

Thanks
Terry

**Comment by terrymacdonald** [comment]
Hi @Smoukus

Steam doesn't have a simple database of games. The data is placed all
around the file system, and each new feature results in a change to the
file system, often a new file. I will need to spend time reverse engineer
the file changes, and the DisplayMagician stream parser will need to be
updated to  handle the various types of entries to pull out the needed data.

I'd Steam just placed that information in the exact same place then it
would already be working.

Hope that clarifies

Thanks
Terry

**Comment by terrymacdonald** [comment]
I've definitely not forgotten this though. It's on my list of things to
work on to get into v2.6.0.

Thanks
Terry

**Comment by terrymacdonald** [comment]
Hi @Smokus,

Unfortunately not. I've just had a breakthrough last night with
DisplayGrids issue that I was experiencing, which means I am able to
shortly start testing the new NVIDIALibrary I've been writing. Maybe you'd
be open to doing some testing for me?

Once that library is ready for testing in my test NVIDIAInfo software, I
can spend some cycles trying to look at Non-steam games for you. That will
no doubt take some time to figure out again. Especially if there are two
copies of the same app that show up in the list as they are viewable in two
libraries. I'll have to think about how to handle that situation.

Thanks

*Terry *

**Comment by terrymacdonald** [comment]
Yep I get it. You want this functionality :).

It takes quite a long time to build this sort of feature though.  Not only
do I have to reverse engineer another application, I have to reverse
engineer the format that the data is stored in. I then have to extend and
build a data processing function that can transform the stored Steam data
into data that we can use, and then I have to display it to the user. That
all takes a lot of time, and I don't really have that much time to offer
for a project that is just a fun project.

I have started on this journey, but I got an issue which forced me to
upgrade the application from .net framework v4.8.1 to .net 8.0. This is a
major upgrade, and it meant quite a bit of time trying to sort out upgrade
errors. I have mostly gotten this part completed, but I'm sure there are
some bugs in there yet to be discovered.

All of this is just to say that I understand this could be useful to have,
and I have started looking into this, but it will be another month or two
before I have something ready enough for you to test.

I'll definitely take you up on the offer though.

Thanks
Terry

**Comment by terrymacdonald** [comment]
OK - Well this has taken ages to do. This addition of Non-steam games ballooned into a large amount of work after I hit problem after problem that I had to solve. I've ended up changing so much of DisplayMagician that this should be considered to be a pre-alpha version of the software.

I'm releasing it here so that you can test out the Non-steam game functionality. But please be warned that there will be a LOT of bugs in this. 

I STRONGLY recommend backing up your previous DisplayMagician folder and saving a copy of it for if you want to revert back. This version of software changes quite a few things.

Over the past month I've:

- Converted the project to .net v8.0 instead of .net Framework v4.8.1
- Ported CoreAudio library over to .net 8.0 so that I can use it in this application
- Created a new Windows Installer from scratch
- Created a new better process detection functionality to detect non-steam games running
- Replaced various libraries that don't support .net 8.0
- Realigned the UI components due to .net 8.0 using different default fonts
- Removed the old incompatible Sharpshell based desktop background context  menu, and wrote new functionality using Windows 10 registry based context menus (still not 100% finished).
- Partially worked on new NVIDIALibrary (still not finished)

This pre-alpha build is DisplayMagician v2.6.0.38073, and it is the first somewhat stable version of DisplayMagician that includes the changes you wanted. You will probably find many issues, but at least we have something.

[DisplayMagicianSetup-v2.6.0.38073.zip](https://github.com/user-attachments/files/16043856/DisplayMagicianSetup-v2.6.0.38073.zip)

Can you please test to make sure that the Non-Steam game stuff works? If it does please report back here, and I will close this issue.

If you find other things wrong with this pre-alpha DisplayMagician, please log a new issue, so that I can keep track of it. I expect there could be issues with the new Windows Installer, as this is a whole new way of installing DisplayMagician.

Anyway - see how you go.

Thanks
Terry

**Comment by terrymacdonald** [comment]
Hi All,

Time for an update.

The fix is in the code and works (also we now support 3rd apps and games added to Steam library too!) but my current hold up is actually in the packaging of the application into a windows installer. I've been testing different ways to do this as the move from .net framework 4.8.1 to .net core 8.0.0 has been a massive change. It has resulted in a completely different set of binaries and dlls and other files being created, as well as a different build process too. This is all taking some time to figure out.

I've gone from a WIXv3 based installer to a windows default installer to an NSIS installer then back to a WIXv4 installer all within the last month. It's necessitated me building a brand new way of handling the desktop background context menu, and even down to which libraries are used within the application. It's been a ton of work.

I'm now down to the last few bugs to iron out, and to finish the new packaging process. I think I should have something ready for everyone to test within the next month (assuming there aren't anymore show stoppers.

My goal is to then send out messages through this issues list to get you all to test some DM v2.6.0 test builds, and then once the major bugs are ironed out I will push out the DM v2.6.0 beta to those users who opted in the beta builds. And then I'll finally push the DM v2.6.0 update to the main user group.

Lastly, I've decided to hold back the NVIDIALibrary update for now until DM v2.7.0 as it requires a LOT more work testing and troubleshooting many errors. The Steam beta client changing file structures has forced my hand here a little bit, so a better NVIDIALibrary will just have to wait for now.

Thanks
Terry

**Comment by terrymacdonald** [comment]
Hi!

This may be fixed by the brand new version of DisplayMagician I just released. I would suggest that you try to the new DisplayMagician v2.7.0-alpha.1 test build available here: 
 https://github.com/terrymacdonald/DisplayMagician/releases/tag/v2.7.0-alpha.1

It should fix your problem, but it may also cause other problems instead. This version has had a complete backend change, new installer, new features, and so it may have some bugs that need squashing. 

Please test it out. and if you find any new problems please log a new issue. If it fixes this issue then please update this issue to tell me and then close this issue (it helps me keep track of what I've still got to fix).

Thanks
Terry

**Comment by LepaBrenaa** [comment]
Not sure what has happened since v2.7.0-alpha.1 and v2.7.2, but DisplayMagician gets completely broken after trying to add Non-Steam Game Shortcut:

<img width="1327" height="890" alt="Image" src="https://github.com/user-attachments/assets/5968a6f6-8d1e-40ea-8cff-dd77046cb11b" />

Here are the log contents after clicking on `Details` in the error message:
```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.NullReferenceException: Object reference not set to an instance of an object.
   at DisplayMagician.UIForms.ShortcutILVRenderer.DrawItem(Graphics g, ImageListViewItem item, ItemState state, Rectangle bounds)
   at Manina.Windows.Forms.ImageListView.ImageListViewRenderer.RenderItems(Graphics g)
   at Manina.Windows.Forms.ImageListView.ImageListViewRenderer.Render(Graphics graphics)
   at Manina.Windows.Forms.ImageListView.OnPaint(PaintEventArgs e)
   at System.Windows.Forms.Control.PaintWithErrorHandling(PaintEventArgs e, Int16 layer)
   at System.Windows.Forms.Control.WmPaint(Message& m)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(HWND hWnd, MessageId msg, WPARAM wparam, LPARAM lparam)


************** Loaded Assemblies **************
System.Private.CoreLib
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Private.CoreLib.dll
----------------------------------------
DisplayMagician
    Assembly Version: 2.7.0.0
    Location: 
----------------------------------------
System.Runtime
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Runtime.dll
----------------------------------------
NLog
    Assembly Version: 5.0.0.0
    Location: 
----------------------------------------
netstandard
    Assembly Version: 2.1.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\netstandard.dll
----------------------------------------
Microsoft.Win32.Registry
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\Microsoft.Win32.Registry.dll
----------------------------------------
DisplayMagician.Shared
    Assembly Version: 2.7.0.0
    Location: 
----------------------------------------
System.Threading.Thread
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Threading.Thread.dll
----------------------------------------
McMaster.Extensions.CommandLineUtils
    Assembly Version: 4.1.1.0
    Location: 
----------------------------------------
System.Net.Http
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Net.Http.dll
----------------------------------------
System.Windows.Forms
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\8.0.25\System.Windows.Forms.dll
----------------------------------------
System.Console
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Console.dll
----------------------------------------
System.Net.ServicePoint
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Net.ServicePoint.dll
----------------------------------------
System.Net.Security
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Net.Security.dll
----------------------------------------
System.ComponentModel
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.ComponentModel.dll
----------------------------------------
Microsoft.Toolkit.Uwp.Notifications
    Assembly Version: 7.1.0.0
    Location: 
----------------------------------------
System.ComponentModel.Primitives
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.ComponentModel.Primitives.dll
----------------------------------------
System.Windows.Forms.Primitives
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\8.0.25\System.Windows.Forms.Primitives.dll
----------------------------------------
System.Drawing.Primitives
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Drawing.Primitives.dll
----------------------------------------
System.Collections.Specialized
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Collections.Specialized.dll
----------------------------------------
System.Collections
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Collections.dll
----------------------------------------
System.Threading
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Threading.dll
----------------------------------------
System.ComponentModel.TypeConverter
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.ComponentModel.TypeConverter.dll
----------------------------------------
System.Runtime.InteropServices
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Runtime.InteropServices.dll
----------------------------------------
System.Diagnostics.StackTrace
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Diagnostics.StackTrace.dll
----------------------------------------
System.Private.Uri
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Private.Uri.dll
----------------------------------------
System.Net.Primitives
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Net.Primitives.dll
----------------------------------------
System.Diagnostics.Tracing
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Diagnostics.Tracing.dll
----------------------------------------
System.Diagnostics.DiagnosticSource
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Diagnostics.DiagnosticSource.dll
----------------------------------------
System.Security.Cryptography
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Security.Cryptography.dll
----------------------------------------
System.Linq
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Linq.dll
----------------------------------------
System.Security.Principal.Windows
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Security.Principal.Windows.dll
----------------------------------------
System.IO.Pipes.AccessControl
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.IO.Pipes.AccessControl.dll
----------------------------------------
System.IO.Pipes
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.IO.Pipes.dll
----------------------------------------
System.Security.AccessControl
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Security.AccessControl.dll
----------------------------------------
System.Security.Claims
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Security.Claims.dll
----------------------------------------
Microsoft.Win32.Primitives
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\Microsoft.Win32.Primitives.dll
----------------------------------------
System.Threading.Overlapped
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Threading.Overlapped.dll
----------------------------------------
System.Text.Encoding.Extensions
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Text.Encoding.Extensions.dll
----------------------------------------
Microsoft.Windows.SDK.NET
    Assembly Version: 10.0.19041.24
    Location: 
----------------------------------------
WinRT.Runtime
    Assembly Version: 2.0.0.0
    Location: 
----------------------------------------
System.Collections.Concurrent
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Collections.Concurrent.dll
----------------------------------------
System.Diagnostics.Process
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Diagnostics.Process.dll
----------------------------------------
System.Drawing.Common
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\8.0.25\System.Drawing.Common.dll
----------------------------------------
System.IO.FileSystem
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.IO.FileSystem.dll
----------------------------------------
System.Security.Cryptography.Algorithms
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Security.Cryptography.Algorithms.dll
----------------------------------------
System.Security.Cryptography.Primitives
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Security.Cryptography.Primitives.dll
----------------------------------------
System.Reflection.Emit
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Reflection.Emit.dll
----------------------------------------
System.Reflection.Emit.ILGeneration
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Reflection.Emit.ILGeneration.dll
----------------------------------------
DynamicComActivator
    Assembly Version: 0.0.0.0
    Location: 
----------------------------------------
System.Collections.NonGeneric
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Collections.NonGeneric.dll
----------------------------------------
System.Memory
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Memory.dll
----------------------------------------
WinCopies.WindowsAPICodePack.Win32Native
    Assembly Version: 2.0.0.6
    Location: 
----------------------------------------
Newtonsoft.Json
    Assembly Version: 13.0.0.0
    Location: 
----------------------------------------
System.Runtime.Serialization.Formatters
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Runtime.Serialization.Formatters.dll
----------------------------------------
System.Diagnostics.TraceSource
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Diagnostics.TraceSource.dll
----------------------------------------
System.Linq.Expressions
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Linq.Expressions.dll
----------------------------------------
System.Runtime.Numerics
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Runtime.Numerics.dll
----------------------------------------
System.ObjectModel
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.ObjectModel.dll
----------------------------------------
System.Runtime.Serialization.Primitives
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Runtime.Serialization.Primitives.dll
----------------------------------------
System.Data.Common
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Data.Common.dll
----------------------------------------
System.Xml.ReaderWriter
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Xml.ReaderWriter.dll
----------------------------------------
System.Private.Xml
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Private.Xml.dll
----------------------------------------
System.Reflection.Emit.Lightweight
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Reflection.Emit.Lightweight.dll
----------------------------------------
System.Reflection.Primitives
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Reflection.Primitives.dll
----------------------------------------
Anonymously Hosted DynamicMethods Assembly
    Assembly Version: 0.0.0.0
    Location: 
----------------------------------------
System.IO.FileSystem.Watcher
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.IO.FileSystem.Watcher.dll
----------------------------------------
System.Net.Mail
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Net.Mail.dll
----------------------------------------
System.ComponentModel.Annotations
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.ComponentModel.Annotations.dll
----------------------------------------
System.Threading.Tasks.Parallel
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Threading.Tasks.Parallel.dll
----------------------------------------
ValveKeyValue
    Assembly Version: 0.11.0.0
    Location: 
----------------------------------------
System.Text.RegularExpressions
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Text.RegularExpressions.dll
----------------------------------------
Interop.IWshRuntimeLibrary
    Assembly Version: 1.0.0.0
    Location: 
----------------------------------------
mscorlib
    Assembly Version: 4.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\mscorlib.dll
----------------------------------------
IconLib
    Assembly Version: 0.73.0.0
    Location: 
----------------------------------------
IconExtractor
    Assembly Version: 1.0.3.0
    Location: 
----------------------------------------
System.Drawing
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\8.0.25\System.Drawing.dll
----------------------------------------
System.Numerics.Vectors
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Numerics.Vectors.dll
----------------------------------------
System.Reflection.Metadata
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Reflection.Metadata.dll
----------------------------------------
System.Collections.Immutable
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Collections.Immutable.dll
----------------------------------------
PresentationCore
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\8.0.25\PresentationCore.dll
----------------------------------------
WindowsBase
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\8.0.25\WindowsBase.dll
----------------------------------------
System.Xaml
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\8.0.25\System.Xaml.dll
----------------------------------------
DirectWriteForwarder
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\8.0.25\DirectWriteForwarder.dll
----------------------------------------
System.Runtime.Extensions
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Runtime.Extensions.dll
----------------------------------------
System.Diagnostics.Debug
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Diagnostics.Debug.dll
----------------------------------------
System.Runtime.CompilerServices.VisualC
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Runtime.CompilerServices.VisualC.dll
----------------------------------------
System.Runtime.CompilerServices.Unsafe
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Runtime.CompilerServices.Unsafe.dll
----------------------------------------
Snippets
    Assembly Version: 0.0.0.0
    Location: 
----------------------------------------
System.Configuration.ConfigurationManager
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\8.0.25\System.Configuration.ConfigurationManager.dll
----------------------------------------
System.Net.WebClient
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Net.WebClient.dll
----------------------------------------
System.IO.Packaging
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\8.0.25\System.IO.Packaging.dll
----------------------------------------
NHotkey
    Assembly Version: 3.0.0.0
    Location: 
----------------------------------------
NHotkey.WindowsForms
    Assembly Version: 3.0.0.0
    Location: 
----------------------------------------
System.ComponentModel.EventBasedAsync
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.ComponentModel.EventBasedAsync.dll
----------------------------------------
Accessibility
    Assembly Version: 4.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\8.0.25\Accessibility.dll
----------------------------------------
Microsoft.Win32.SystemEvents
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\8.0.25\Microsoft.Win32.SystemEvents.dll
----------------------------------------
System.Resources.Extensions
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App\8.0.25\System.Resources.Extensions.dll
----------------------------------------
System.Runtime.Loader
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Runtime.Loader.dll
----------------------------------------
AudioSwitcher.AudioApi.CoreAudio
    Assembly Version: 4.0.0.308
    Location: 
----------------------------------------
AudioSwitcher.AudioApi
    Assembly Version: 4.0.0.308
    Location: 
----------------------------------------
PInvokeDelegateFactoryInternalAssembly
    Assembly Version: 0.0.0.0
    Location: 
----------------------------------------
EDIDParser
    Assembly Version: 1.2.5.3
    Location: 
----------------------------------------
AutoUpdater.NET
    Assembly Version: 1.9.2.0
    Location: 
----------------------------------------
System.Web.HttpUtility
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Web.HttpUtility.dll
----------------------------------------
System.Net.Requests
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Net.Requests.dll
----------------------------------------
System.Xml.XmlSerializer
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Xml.XmlSerializer.dll
----------------------------------------
System.Net.WebHeaderCollection
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Net.WebHeaderCollection.dll
----------------------------------------
System.Net.Sockets
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Net.Sockets.dll
----------------------------------------
System.Net.NameResolution
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Net.NameResolution.dll
----------------------------------------
System.Threading.ThreadPool
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Threading.ThreadPool.dll
----------------------------------------
System.Runtime.Intrinsics
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.Runtime.Intrinsics.dll
----------------------------------------
Microsoft.CSharp
    Assembly Version: 8.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\Microsoft.CSharp.dll
----------------------------------------
ImageListView
    Assembly Version: 13.8.2.0
    Location: 
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Location: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\8.0.25\System.dll
----------------------------------------

************** JIT Debugging **************

```

As soon as I click on `Game Shortcuts` after opening DisplayMagician, the error above pops up and I can't make any new shortcuts.
My DisplayMagician is not broken until I reinstall it.

Here's the Support Zip that I just generate via DisplayMagician with Full Application Trace logging:
[DisplayMagician-Support-20260606-1358.zip](https://github.com/user-attachments/files/28666385/DisplayMagician-Support-20260606-1358.zip)

**Comment by terrymacdonald** [comment]
Hi, known issue. Should be fixed already in DMv3 codebase. There was a bug with a missed image being saved, which has been fixed in code, but not part of a release yet.

**Comment by terrymacdonald** [comment]
Please see #387 as I think that's the one that's affecting you.

**Comment by LepaBrenaa** [comment]
Thanks for getting back to me quickly. If possible, I'd like to be your tester for v3 codebase.
